### PR TITLE
[9.0][website_portal_contact] Manage your contacts from your portal

### PR DESCRIPTION
**Pull request by yajo**
This module extends the functionality of the website portal to allow your users to manage their contacts.

To use this module, you need to:
1. Log in as a portal user.
2. Go to your portal home.
3. Go to your contacts manager.
4. Read, create, edit or delete (disable) any contacts you want!

Known issues / Roadmap:
- This uses the backported version of the website portal from v10. When migrating to v10, we should adapt to any changes there.
- Remove workarounds for https://github.com/odoo/odoo/issues/12961 in JS tour when fixed.

@Tecnativa

### CODE REVIEW

**Comment by yajo** [comment]
Please could anybody help me here? If I log into the runbot's all database as portal user, I get access error:

```
403 Forbidden:

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/website/models/ir_http.py", line 243, in _handle_exception
    response = super(ir_http, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/base/ir/ir_http.py", line 157, in _handle_exception
    return request._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 785, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/base/ir/ir_http.py", line 182, in _dispatch
    result = request.dispatch()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 844, in dispatch
    r = self._call_function(**self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 320, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 313, in checked_call
    result = self.endpoint(*a, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 963, in __call__
    return self.method(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/http.py", line 513, in response_wrap
    response = f(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/website_portal_contact/controllers/main.py", line 104, in account
    self._contacts_domain())
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 490, in new_api
    result = method(self._model, cr, uid, *args, **old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/models.py", line 1634, in search_count
    res = self.search(cr, user, args, context=context, count=True)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/models.py", line 1658, in search
    return self._search(cr, user, args, offset=offset, limit=limit, order=order, context=context, count=count)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/base/res/res_partner.py", line 622, in _search
    count=count, access_rights_uid=access_rights_uid)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/models.py", line 4770, in _search
    self.check_access_rights(cr, access_rights_uid or user, 'read')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/models.py", line 3552, in check_access_rights
    return self.pool.get('ir.model.access').check(cr, uid, self._name, operation, raise_exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in check
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/tools/cache.py", line 85, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3151351-239-0d445f/openerp/addons/base/ir/ir_model.py", line 847, in check
    raise openerp.exceptions.AccessError(msg % msg_params)
AccessError: (u'Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Other Extra Rights/Public\n\t- Other Extra Rights/Portal\n\t- Other Extra Rights/Contact Creation\n\t- Human Resources/Employee\n\t- Other Extra Rights/Portal\n\t- Sales/Manager\n\t- Sales/User: Own Leads Only\n\n(Document model: res.partner)', None)
```

If I install the module on `base` database, and I insall `portal` to create the demo portal user, and log in as it, it works fine (as in localhost).

So I guess some module must be interferring with the permissions management, but don't know which. I compared `ir.model.access` and `ir.rules` with mine in localhost and seem equal...

**Comment by yajo** [comment]
Don't believe Travis: it's affected by https://github.com/odoo/odoo/issues/12964. This is failing due to the last comment.

**Comment by yajo** [comment]
Continuing with my public monologue, the solution to https://github.com/OCA/website/pull/239#issuecomment-236184452 is removing `website_seo_redirection`. Seems related to https://github.com/OCA/website/issues/222#issuecomment-230632970. I'm on it.

**Comment by yajo** [comment]
I cherry-picked fix from https://github.com/OCA/website/pull/240

**Comment by yajo** [comment]
Rebased and removed cherrypick now that #241 is merged.

File: website_portal_contact/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        values = self._contacts_clean_values(kwargs)
+        _logger.debug("Updating %r with: %s", contact, values)
+        contact.write(values)
+        return local_redirect(redirect.format(contact.id))
+
+    @route("/my/contacts/<model('res.partner'):contact>/disable",
+           auth="user", website=True)
+    def portal_my_contacts_disable(self, contact, redirect="/my/contacts"):
+        """Disable a contact."""
+        _logger.debug("Disabilng %r", contact)
```
Comment (pedrobaeza) [inline website_portal_contact/controllers/main.py]: s/Disabilng/Disabling

File: website_portal_contact/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+#. module: website_portal_contact
+#: model:ir.ui.view,arch_db:website_portal_contact.contacts_followup
+msgid "Edit Contact"
+msgstr "Editar contacto"
+
+#. module: website_portal_contact
+#: model:ir.ui.view,arch_db:website_portal_contact.contacts_table
+msgid "Email"
+msgstr "Email"
```
Comment (pedrobaeza) [inline website_portal_contact/i18n/es.po]: Correo electrónico

File: website_portal_contact/i18n/website_portal_contact.pot [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline website_portal_contact/i18n/website_portal_contact.pot]: Don't include POT file

**Comment by pedrobaeza** [comment]
Functional test :+1:

**Comment by rafaelbn** [comment]
Tested functionaly :+1:

**Pull request merged**